### ISSUE TITLE
[QUESTION] Information on NER component

### ISSUE THREAD

**Issue by rom1K** [CAMeL-Lab/camel_tools#70]
**Describe what you would like to know about CAMeL Tools.**

Hello, I wanted to know if you could provide some information regarding the NER component of the library.

In the catalog JSON file, you mention that you are using a finetuned AraBERT model, with the specified version being 1.0.0. So from here, I wanted to know:
- whether the model used as base was indeed AraBERTv1 from [this repo](https://github.com/aub-mind/arabert) ?
- which dataset you used ?
- whether you used the FARASA preprocessing for the finetuning or your own given that they used the former for pretraining ?

I ask because while doing some research I saw that your lab has produced multiple arabic BERT models, which have the benefit of:
- having used the `camel_tools` preprocessing rather the FARASA for both pretraining and finetuning
- have dialect-specific variants, which may be interesting in some cases
- seem to outperform the AraBERTv1 on NER tasks according to your paper

I was wondering whether you would consider making these models available for use in this library ? I know you have released the code and pretrained model, and I am planning on experimenting with this, but thought it would be a nice addition.

**Comment by balhafni** [comment]
So for the NER component, we didn't do any preprocessing before fine-tuning and we used `aubmindlab/bert-base-arabertv01`, which did not use FARASA segmentation before the pretraining. 

We also just released new NER models which were fine-tuned using our own CAMeLBERT models on Hugging Face's [model hub](https://huggingface.co/CAMeL-Lab). Here's an [example](https://huggingface.co/CAMeL-Lab/bert-base-arabic-camelbert-msa-ner) on how to use the CAMeLBERT NER MSA model.  **Disclaimer**: Although in the [example](https://huggingface.co/CAMeL-Lab/bert-base-arabic-camelbert-msa-ner) we use the NER component from CAMeL Tools to load the model directly from hub, this is still a work in progress so please use with caution.

**Comment by rom1K** [comment]
Okay I see, thanks a lot for your answers, you've been a great help :)

I have one last question, pertaining to the NER models you released during the week-end : in the pre-processing section of the paper you mention that you apply the following:

1. Remove invalid characters, normalize whitespaces, and remove lines without arabic characters
2. Remove diacritics and kashida using CAMeL Tools 
3. Split lines into sentences.

However in the example documentation it seems you are only applying a simple tokenization scheme to the data. I wanted to know whether this was because the arabic in the example was already normalized according to the procedure above, or if for NER you recommend a different normalization then for the pretraining MLM task ?

**Comment by balhafni** [comment]
Good question! The `simple_word_tokenize()` that's used in the example only does whitespace and punctuation tokenizations. This is basically needed to ensure that our NER component gets a list of strings (words) and outputs a list of labels (one label per word). 

As you mentioned, our CAMeLBERT tokenizers where pretrained on undiacritized clean text, so we would expect the input to be undiacritized when we feed it to any of the CAMeLBERT models (including the models which were fine-tuned using CAMeLBERT such as NER). The sentence in the example referenced in my previous comment is clean and it doesn't require any undiacritization. At the moment, we don't do undiacritization internally as a preprocessing step in the CAMeL Tools NER component (although we probably should).

**Comment by rom1K** [comment]
I see, that's what I was inferring from the example text, but as I don't know how to read Arabic so wasn't 100% sure ^^'

In all cases thanks again for your help ! I'm going to be using this library and some of your models, so the information you gave me will definitely be of use. I'll be looking forward to future work from your lab - and if I see anything I can do regarding this repo I'll be sure to come back with a PR :)

**Comment by balhafni** [comment]
Always happy to help!
